lib/msf/core/web_services/authentication/strategies/api_token.rb

```diff
@@ -48,6 +53,10 @@ def validate_user(user)
         !user.nil? ? {valid: true, code: 0, message: nil} : {valid: false, code: 401, message: "Invalid API token."}
       end
 
+      # Authenticates the API token from a configuration file
+      def auth_from_file(token)
+        token == request.env['msf.token_from_file'] ? success!(message: "Successful auth from file token") : fail!
```

Comment:
Is there a reason `fail!` was used rather than a `throw(:warden,...)` as used in the `auth_from_db` method? The later would allow control the prepended to the "Authenticate to access this resource." error message. See the `post_unauthenticated` method in `lib/msf/core/web_services/servlet/auth_servlet.rb`.